### DISCUSSION TITLE
Public URL for private files

### DISCUSSION THREAD

**Discussion by netsinerji** [directus/directus#15852] [Archive]
Hi!
Is it possible to allow direct URL access to files which are not open to public?
I need to use https://mydomain.com/assets/{id} as image source, but when I allow public role to view files, they can view the full file list from https://mydomain.com/files address.
It could be a good idea to allow limited "search" or "query" permission to a specific role, rather than view all, so that public users can retrieve an item only if they provide an unguessable ID.

**Comment by azrikahar** [comment]
> but when I allow public role to view files, they can view the full file list from https://mydomain.com/files address.

Rather than allowing public role to have full read access to `directus_files` (assuming it wasn't restricted in your setup), you can also create a folder named "public" for example, than only allow read permission to files that are _within_ said folder. However this of course still allows them to view the list of files within that folder via the `/files` endpoint, albeit not the full list.

> Is it possible to allow direct URL access to files which are not open to public?

You can create a dedicated role and user with a static access token, and use the access token in those URLs. This will be the similar approach used in the Static Access Token option available in WYSIWYG fields.

> It could be a good idea to allow limited "search" or "query" permission to a specific role, rather than view all, so that public users can retrieve an item only if they provide an unguessable ID.

There's a discussion revolving around this idea over here: #11464, so feel free to give it an upvote 👍

**Reply by netsinerji** [reply]
Thanks for your support, much appreciated.
Using a static access token seems to be a reasonable solution for me.
